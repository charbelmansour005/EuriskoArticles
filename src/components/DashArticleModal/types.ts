export type ArticleCardModalProps = {
  headline?: string
  leadParagraph?: string
  author?: string
  url?: string | undefined | any
  section?: string | null
  multimedia?: {url: string}[] | any
  hideModal?: () => void
}
