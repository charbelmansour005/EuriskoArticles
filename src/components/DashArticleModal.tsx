import {StyleSheet, Text, View, Linking, Image} from 'react-native';
import React from 'react';
import Logo from '../../assets/eurisko.jpg';
import {themeColors} from '../helpers/themeColors';
import {TouchableRipple} from 'react-native-paper';

type ArticleCardModalProps = {
  headline?: string;
  leadParagraph?: string;
  author?: string;
  url?: string | undefined | any;
  section?: string | null;
  hideModal?: () => void;
};

const DashArticleModal = ({...props}: ArticleCardModalProps) => {
  return (
    <View style={styles.modalArticleView}>
      <View style={styles.logoContainer}>
        <Image source={Logo} style={styles.logoModal} />
        <TouchableRipple
          onPress={props.hideModal}
          rippleColor={themeColors.salmon}
          style={styles.modalButton}>
          <Text style={{color: 'black', fontWeight: 'bold'}}>
            {`Return to Dashboard`}
          </Text>
        </TouchableRipple>
      </View>
      <Text
        selectable={true}
        style={{letterSpacing: 1, ...styles.modalArticleHeadline}}>
        {props.headline}
      </Text>
      <Text
        selectable={true}
        style={{
          color: themeColors.pitchblack,
          ...styles.modalArticleElements,
        }}>
        {props.author}
      </Text>
      <Text
        style={{
          color: themeColors.pitchblack,
          ...styles.modalArticleElements,
        }}>
        {props.section}
      </Text>
      <Text
        onPress={() => Linking.openURL(props.url)}
        style={{
          color: themeColors.salmon,
          fontWeight: 'bold',
          ...styles.modalArticleElements,
        }}>
        View on Website
      </Text>
      <Text selectable={true} style={styles.modalArticleParagraph}>
        {props.leadParagraph}
      </Text>
    </View>
  );
};

export default DashArticleModal;

const styles = StyleSheet.create({
  modalArticleView: {
    top: 0,
    bottom: 0,
    left: 0,
    right: 0,
    position: 'relative',
    marginVertical: '20%',
  },
  logoContainer: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
  },
  logoModal: {
    height: 120,
    width: 110,
    borderRadius: 3,
    borderWidth: 4,
    borderColor: 'black',
  },
  modalArticleHeadline: {
    fontWeight: 'bold',
    color: themeColors.salmon,
    padding: 10,
    fontSize: 15,
    borderTopWidth: 1,
    borderColor: themeColors.lightgray,
    textAlign: 'left',
  },
  modalArticleElements: {
    padding: 10,
    fontSize: 15,
    borderBottomWidth: 1,
    borderColor: themeColors.lightgray,
    textAlign: 'left',
  },
  modalArticleParagraph: {
    color: 'black',
    padding: 20,
    borderColor: themeColors.lightgray,
    borderBottomWidth: 1,
    borderRadius: 0,
    fontSize: 15,
    lineHeight: 25,
    fontWeight: 'bold',
  },
  modalButton: {
    padding: 10,
    alignItems: 'center',
    marginVertical: 20,
    borderRadius: 5,
  },
});
