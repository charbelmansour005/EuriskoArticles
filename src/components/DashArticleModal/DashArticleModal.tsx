// react / native
import React from 'react'
import {StyleSheet, Text, View, Linking, Image} from 'react-native'
import {TouchableRipple} from 'react-native-paper'
// logo + types + helper
import {themeColors} from '../../helpers/themeColors'
import Logo from '../../../assets/eurisko.jpg'
import {ArticleCardModalProps} from './types'

const DashArticleModal = ({...props}: ArticleCardModalProps) => {
  console.log(props.multimedia)

  const IMG_URL = 'https://static01.nyt.com/'

  return (
    <View style={styles.modalArticleView} testID="parent">
      <TouchableRipple
        testID="modalBackButton"
        onPress={props?.hideModal}
        rippleColor={themeColors.lightgray}
        style={styles.modalButton}>
        <Text
          testID="modalBackText"
          style={{color: 'black', fontWeight: 'bold'}}>
          Return
        </Text>
      </TouchableRipple>
      <View style={styles.logoContainer} testID="logoContainer">
        {props?.multimedia[0]?.url?.length ? (
          <Image
            source={{uri: IMG_URL + props.multimedia[0].url}}
            style={styles.ImageModal}
          />
        ) : (
          <Image testID="image" source={Logo} style={styles.logoModal} />
        )}
      </View>
      <Text
        testID="modalHeadline"
        selectable={true}
        style={{letterSpacing: 1, ...styles.modalArticleHeadline}}>
        {props?.headline}
      </Text>
      <Text
        testID="modalAuthor"
        selectable={true}
        style={{
          color: themeColors.pitchblack,
          ...styles.modalArticleElements,
        }}>
        {props.author}
      </Text>
      <Text
        testID="modalSection"
        style={{
          color: themeColors.pitchblack,
          ...styles.modalArticleElements,
        }}>
        {props?.section}
      </Text>
      <Text
        testID="webLink"
        onPress={() => Linking.openURL(props?.url)}
        style={{
          color: themeColors.darkgray,
          fontWeight: 'bold',
          ...styles.modalArticleElements,
        }}>
        View on Website
      </Text>
      <Text
        testID="leadParagraph"
        selectable={true}
        style={styles.modalArticleParagraph}>
        {props?.leadParagraph}
      </Text>
      {/* <Image source={{uri: `https://www.nytimes.com/${props.image}`}} /> */}
    </View>
  )
}

export default DashArticleModal

const styles = StyleSheet.create({
  modalArticleView: {
    top: 0,
    bottom: 0,
    left: 0,
    right: 0,
    position: 'relative',
    marginVertical: '20%',
  },
  logoContainer: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
  },
  logoModal: {
    height: 110,
    width: 105,
    borderRadius: 3,
    borderWidth: 0,
    borderColor: 'black',
    marginBottom: 20,
  },
  ImageModal: {
    height: 150,
    width: 150,
    borderRadius: 3,
    borderWidth: 0,
    borderColor: 'black',
    marginBottom: 20,
  },
  modalArticleHeadline: {
    fontWeight: 'bold',
    color: themeColors.darkgray,
    padding: 10,
    fontSize: 15,
    borderTopWidth: 1,
    borderColor: themeColors.lightgray,
    textAlign: 'center',
  },
  modalArticleElements: {
    padding: 10,
    fontSize: 15,
    borderTopWidth: 1,
    borderColor: themeColors.lightgray,
    textAlign: 'center',
  },
  modalArticleParagraph: {
    color: 'black',
    padding: 20,
    borderColor: themeColors.lightgray,
    borderBottomWidth: 1,
    borderRadius: 0,
    fontSize: 15,
    lineHeight: 25,
    fontWeight: 'bold',
  },
  modalButton: {
    padding: 10,
    alignItems: 'center',
    marginBottom: 20,
    borderRadius: 5,
  },
})
