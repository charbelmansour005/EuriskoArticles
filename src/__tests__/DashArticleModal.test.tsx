import React from 'react'
import renderer from 'react-test-renderer'
import DashArticleModal from '../components/DashArticleModal/DashArticleModal'

describe('DashArticleModal Component', () => {
  test('Component should display props', async () => {
    const tree = renderer.create(<DashArticleModal />).toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toBeTruthy()
  })
})
