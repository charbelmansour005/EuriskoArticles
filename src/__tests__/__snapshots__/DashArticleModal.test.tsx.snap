// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component should display props 1`] = `
<View
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "marginVertical": "20%",
      "position": "relative",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "center",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/eurisko.jpg",
        }
      }
      style={
        Object {
          "borderColor": "black",
          "borderRadius": 3,
          "borderWidth": 0,
          "height": 120,
          "width": 110,
        }
      }
      testID="image"
    />
    <View
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          false,
          false,
          Object {
            "alignItems": "center",
            "borderRadius": 5,
            "marginVertical": 20,
            "padding": 10,
          },
        ]
      }
      testID="modalBackButton"
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontWeight": "bold",
          }
        }
        testID="modalBackText"
      >
        Return to Dashboard
      </Text>
    </View>
  </View>
  <Text
    selectable={true}
    style={
      Object {
        "borderColor": "lightgray",
        "borderTopWidth": 1,
        "color": "#29a329",
        "fontSize": 15,
        "fontWeight": "bold",
        "letterSpacing": 1,
        "padding": 10,
        "textAlign": "center",
      }
    }
    testID="modalHeadline"
  />
  <Text
    selectable={true}
    style={
      Object {
        "borderColor": "lightgray",
        "borderTopWidth": 1,
        "color": "#0a0a0f",
        "fontSize": 15,
        "padding": 10,
        "textAlign": "center",
      }
    }
    testID="modalAuthor"
  />
  <Text
    style={
      Object {
        "borderColor": "lightgray",
        "borderTopWidth": 1,
        "color": "#0a0a0f",
        "fontSize": 15,
        "padding": 10,
        "textAlign": "center",
      }
    }
    testID="modalSection"
  />
  <Text
    onPress={[Function]}
    style={
      Object {
        "borderColor": "lightgray",
        "borderTopWidth": 1,
        "color": "#29a329",
        "fontSize": 15,
        "fontWeight": "bold",
        "padding": 10,
        "textAlign": "center",
      }
    }
    testID="webLink"
  >
    View on Website
  </Text>
  <Text
    selectable={true}
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "lightgray",
        "borderRadius": 0,
        "color": "black",
        "fontSize": 15,
        "fontWeight": "bold",
        "lineHeight": 25,
        "padding": 20,
      }
    }
    testID="leadParagraph"
  />
</View>
`;
